builtin/aws/lambda/ecs.go

```diff
@@ -0,0 +1,381 @@
+package lambda
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/builtin/aws/utils"
+)
+
+// This launches an ECS task to run the given image. It creates an ECS cluster configured
+// against Fargate for this purpose, as well as creates an IAM role and security group
+// so the task can be accessed via TCP.
+
+type ecsLauncher struct {
+	Region       string
+	PublicKey    string
+	HostKey      string
+	Image        string
+	DeploymentId string
+
+	LogOutput io.Writer
+
+	roleName string
+	roleArn  string
+
+	status terminal.Status
+}
+
+const ecsClusterName = "waypoint-lambda-exec"
+
+// SetupCluster creates an ECS cluster if there isn't one.
+func (e *ecsLauncher) SetupCluster(sess *session.Session, ctx context.Context, log hclog.Logger) error {
+	ecsSvc := ecs.New(sess)
+
+	desc, err := ecsSvc.DescribeClusters(&ecs.DescribeClustersInput{
+		Clusters: []*string{aws.String(ecsClusterName)},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(desc.Clusters) > 1 {
+		log.Info("existing ECS cluster found", "arn", *desc.Clusters[0].ClusterArn)
+		return nil
+	}
+
+	log.Info("creating ECS cluster")
+	out, err := ecsSvc.CreateCluster(&ecs.CreateClusterInput{
+		ClusterName: aws.String(ecsClusterName),
+	})
+
+	if err != nil {
+		log.Error("error creating new cluster", "error", err)
+		return err
+	}
+
+	log.Info("created cluster", "arn", *out.Cluster.ClusterArn)
+
+	return nil
+}
+
+const rolePolicy = `{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+			"Sid": "",
+			"Effect": "Allow",
+			"Principal": {
+				"Service": "ecs-tasks.amazonaws.com"
+			},
+			"Action": "sts:AssumeRole"
+		}
+	]
+}`
+
+// SetupRole creates an IAM role to use for ECS task execution. Reuses a role if there is already one.
+func (e *ecsLauncher) SetupRole(L hclog.Logger, sess *session.Session, log hclog.Logger, app *component.Source) error {
+	svc := iam.New(sess)
+
+	e.roleName = "ecr-" + app.App
+
+	log.Info("setting up IAM role")
+	L.Debug("attempting to retrieve existing role", "role-name", e.roleName)
+
```

Comment:
As stolen from https://github.com/hashicorp/waypoint/blob/68c046a2074f16d63552dfcadcce5745c4a15197/builtin/aws/ecs/platform.go#L406-L410
```suggestion

	// role names have to be 64 characters or less, and the client side doesn't validate this.
	if len(roleName) > 64 {
		roleName = roleName[:64]
	}
	
```